src/saveToCase.js

```diff
@@ -0,0 +1,575 @@
+                    properties: [
+                        "use_update",
+                        "update_property",
+                    ],
+                    isCollapsed: function (mug) {
+                        return !updatesCase(mug);
+                    },
+                },
+                {
+                    slug: "close",
+                    displayName: "Close",
+                    properties: [
+                        "use_close",
+                        "close_condition",
+                    ],
+                    isCollapsed: function (mug) {
+                        return !closesCase(mug);
+                    },
+                },
+                {
+                    slug: "index",
+                    displayName: "Index",
+                    properties: [
+                        "use_index",
+                        "index_property",
+                    ],
+                    isCollapsed: function (mug) {
+                        return !indexesCase(mug);
+                    },
+                },
+            ]
+        };
+
+    $.vellum.plugin("saveToCase", {}, {
+        getAdvancedQuestions: function () {
+            return this.__callOld().concat(["SaveToCase"]);
+        },
+        handleMugParseFinish: function (mug) {
+            this.__callOld();
+            // cases that are created use a setvalue for case_id
+            if (!createsCase(mug)) {
+                return;
+            }
+            var values = _.object(_.map(mug.form.getSetValues(), function (value) {
+                    return [value.ref, value];
+                })),
+                caseIdRegex = new RegExp(mug.p.nodeID + "/case/@case_id$");
+
+            _.each(values, function(value) {
+                if (caseIdRegex.test(value.ref)) {
+                    mug.p.case_id = value.value;
+                }
+            });
+        },
+        getMugTypes: function () {
+            var types = this.__callOld();
+            types.normal.SaveToCase = util.extend(
+                mugs.defaultOptions, saveToCaseMugOptions);
+            return types;
+        },
+        getSections: function (mug) {
+            if (sectionData.hasOwnProperty(mug.__className)) {
+                return _.map(sectionData[mug.__className], function (section) {
+                    var tmpSection = _.clone(section);
+                    if (_.isFunction(tmpSection.isCollapsed)) {
+                        tmpSection.isCollapsed = tmpSection.isCollapsed(mug);
+                    }
+                    return tmpSection;
+                });
+            }
+            return this.__callOld();
+        },
+        parseBindElement: function (form, el, path) {
+            var mug = form.getMugByPath(path);
+            if (!mug) {
+                var casePathRegex = /\/case\/((create|update|index)\/(\w+)|close|@date_modified|@user_id|@case_id)$/,
+                    matchRet = _.compact(path.match(casePathRegex));
```

Comment:
Minor: you could avoid `_.compact` and simplify this a bit by using [non-capturing parentheses `(?:x)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#grouping-back-references). Not sure if you think this is clearer:
```js
var casePathRegex = /\/case\/((?:create|update|index)\/(\w+)|close|@date_modified|@user_id|@case_id)$/,
    match = path.match(casePathRegex);
if (match) {
    ...
    if (match[2]) {
        // path is .../case/(create|update|index)/(NAME), and match[2] is the NAME
        var prop = match[2], ...
    } else {
        // path is .../case/(close|@date_modified|@user_id|@case_id)
        var attr = match[1], ...
    }
}
```